src/main/java/org/spongepowered/mod/mixin/world/MixinWorld.java

```diff
@@ -93,21 +125,106 @@ public BlockLoc getBlock(int x, int y, int z) {
 
     @Override
     public Biome getBiome(Vector3d position) {
-        throw new UnsupportedOperationException();
+        BlockPos blockPos = new BlockPos(position.getX(), position.getY(), position.getZ());
+        return new SpongeBiome(blockPos, getBiomeGenForCoords(blockPos), getChunkFromBlockCoords(blockPos));
     }
 
+    @SuppressWarnings("unchecked")
     @Override
     public Collection<Entity> getEntities() {
-        return new ArrayList<Entity>();
+        return (Collection<Entity>) (Object) loadedEntityList;
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public Collection<Entity> getEntities(Predicate<Entity> filter) {
+        return (Collection<Entity>) (Object) func_175644_a(net.minecraft.entity.Entity.class,
+                (Predicate<net.minecraft.entity.Entity>) (Object) filter);
     }
 
     @Override
     public Optional<Entity> createEntity(EntityType type, Vector3d position) {
+        try {
+            net.minecraft.entity.Entity entity =
+                    (net.minecraft.entity.Entity) type.getEntityClass().getConstructor(net.minecraft.world.World.class).newInstance(this);
+            entity.setPosition(position.getX(), position.getY(), position.getZ());
+            if (spawnEntityInWorld(entity)) {
+                return Optional.of((Entity) entity);
+            }
+        } catch (ReflectiveOperationException ignored) {
```

Comment:
This doesn't compile, see https://travis-ci.org/SpongePowered/Sponge/builds/45614519#L514-L517